// @storybook/html@10.0.8/dist/_browser-chunks/chunk-R4NDMI3K.js
import {
  __export,
  __name
} from "./chunk-JFJ5UJ7Q.js";

// src/entry-preview.ts
var entry_preview_exports = {};
__export(entry_preview_exports, {
  argTypesEnhancers: () => argTypesEnhancers,
  parameters: () => parameters,
  render: () => render,
  renderToCanvas: () => renderToCanvas
});
import { enhanceArgTypes } from "storybook/internal/docs-tools";

// src/render.ts
import { global } from "@storybook/global";
import { simulateDOMContentLoaded, simulatePageLoad } from "storybook/preview-api";
import { dedent } from "ts-dedent";
var { Node } = global;
var render = /* @__PURE__ */ __name((args, context) => {
  const { id, component: Component } = context;
  if (typeof Component === "string") {
    let output = Component;
    Object.keys(args).forEach((key) => {
      output = output.replace(`{{${key}}}`, args[key]);
    });
    return output;
  }
  if (Component instanceof HTMLElement) {
    const output = Component.cloneNode(true);
    Object.keys(args).forEach((key) => {
      output.setAttribute(
        key,
        typeof args[key] === "string" ? args[key] : JSON.stringify(args[key])
      );
    });
    return output;
  }
  if (typeof Component === "function") {
    return Component(args, context);
  }
  console.warn(dedent`
    Storybook's HTML renderer only supports rendering DOM elements and strings.
    Received: ${Component}
  `);
  throw new Error(`Unable to render story ${id}`);
}, "render");
function renderToCanvas({ storyFn, kind, name, showMain, showError, forceRemount }, canvasElement) {
  const element = storyFn();
  showMain();
  if (typeof element === "string") {
    canvasElement.innerHTML = element;
    simulatePageLoad(canvasElement);
  } else if (element instanceof Node) {
    if (canvasElement.firstChild === element && forceRemount === false) {
      return;
    }
    canvasElement.innerHTML = "";
    canvasElement.appendChild(element);
    simulateDOMContentLoaded();
  } else {
    showError({
      title: `Expecting an HTML snippet or DOM node from the story: "${name}" of "${kind}".`,
      description: dedent`
        Did you forget to return the HTML snippet from the story?
        Use "() => <your snippet or node>" or when defining the story.
      `
    });
  }
}
__name(renderToCanvas, "renderToCanvas");

// src/entry-preview.ts
var parameters = { renderer: "html" };
var argTypesEnhancers = [enhanceArgTypes];

export {
  render,
  renderToCanvas,
  parameters,
  argTypesEnhancers,
  entry_preview_exports
};
