// @storybook/web-components@10.0.8/dist/entry-preview-argtypes.js
import {
  getCustomElements,
  isValidComponent,
  isValidMetaData
} from "./_browser-chunks/chunk-CTBWQQZK.js";
import "./_browser-chunks/chunk-NIAEHH3S.js";
import {
  __name
} from "./_browser-chunks/chunk-JFJ5UJ7Q.js";

// src/entry-preview-argtypes.ts
import { enhanceArgTypes } from "storybook/internal/docs-tools";

// src/docs/custom-elements.ts
import { logger } from "storybook/internal/client-logger";
import invariant from "tiny-invariant";
function mapItem(item, category) {
  let type;
  switch (category) {
    case "attributes":
    case "properties":
      type = { name: item.type?.text || item.type };
      break;
    case "slots":
      type = { name: "string" };
      break;
    default:
      type = { name: "void" };
      break;
  }
  return {
    name: item.name,
    required: false,
    description: item.description,
    type,
    table: {
      category,
      type: { summary: item.type?.text || item.type },
      defaultValue: {
        summary: item.default !== void 0 ? item.default : item.defaultValue
      }
    }
  };
}
__name(mapItem, "mapItem");
function mapEvent(item) {
  let name = item.name.replace(/(-|_|:|\.|\s)+(.)?/g, (_match, _separator, chr) => {
    return chr ? chr.toUpperCase() : "";
  }).replace(/^([A-Z])/, (match) => match.toLowerCase());
  name = `on${name.charAt(0).toUpperCase() + name.substr(1)}`;
  return [{ name, action: { name: item.name }, table: { disable: true } }, mapItem(item, "events")];
}
__name(mapEvent, "mapEvent");
function mapData(data, category) {
  return data && data.filter((item) => item && item.name).reduce((acc, item) => {
    if (item.kind === "method") {
      return acc;
    }
    switch (category) {
      case "events":
        mapEvent(item).forEach((argType) => {
          invariant(argType.name, `${argType} should have a name property.`);
          acc[argType.name] = argType;
        });
        break;
      default:
        acc[item.name] = mapItem(item, category);
        break;
    }
    return acc;
  }, {});
}
__name(mapData, "mapData");
var getMetaDataExperimental = /* @__PURE__ */ __name((tagName, customElements) => {
  if (!isValidComponent(tagName) || !isValidMetaData(customElements)) {
    return null;
  }
  const metaData = customElements.tags.find(
    (tag) => tag.name.toUpperCase() === tagName.toUpperCase()
  );
  if (!metaData) {
    logger.warn(`Component not found in custom-elements.json: ${tagName}`);
  }
  return metaData;
}, "getMetaDataExperimental");
var getMetaDataV1 = /* @__PURE__ */ __name((tagName, customElements) => {
  if (!isValidComponent(tagName) || !isValidMetaData(customElements)) {
    return null;
  }
  let metadata;
  customElements?.modules?.forEach((_module) => {
    _module?.declarations?.forEach((declaration) => {
      if (declaration.tagName === tagName) {
        metadata = declaration;
      }
    });
  });
  if (!metadata) {
    logger.warn(`Component not found in custom-elements.json: ${tagName}`);
  }
  return metadata;
}, "getMetaDataV1");
var getMetaData = /* @__PURE__ */ __name((tagName, manifest) => {
  if (manifest?.version === "experimental") {
    return getMetaDataExperimental(tagName, manifest);
  }
  return getMetaDataV1(tagName, manifest);
}, "getMetaData");
var extractArgTypesFromElements = /* @__PURE__ */ __name((tagName, customElements) => {
  const metaData = getMetaData(tagName, customElements);
  return metaData && {
    ...mapData(metaData.members ?? [], "properties"),
    ...mapData(metaData.properties ?? [], "properties"),
    ...mapData(metaData.attributes ?? [], "attributes"),
    ...mapData(metaData.events ?? [], "events"),
    ...mapData(metaData.slots ?? [], "slots"),
    ...mapData(metaData.cssProperties ?? [], "css custom properties"),
    ...mapData(metaData.cssParts ?? [], "css shadow parts")
  };
}, "extractArgTypesFromElements");
var extractArgTypes = /* @__PURE__ */ __name((tagName) => {
  const cem = getCustomElements();
  return extractArgTypesFromElements(tagName, cem);
}, "extractArgTypes");
var extractComponentDescription = /* @__PURE__ */ __name((tagName) => {
  const metaData = getMetaData(tagName, getCustomElements());
  return metaData && metaData.description;
}, "extractComponentDescription");

// src/entry-preview-argtypes.ts
var parameters = {
  docs: {
    extractArgTypes,
    extractComponentDescription
  }
};
var argTypesEnhancers = [enhanceArgTypes];
export {
  argTypesEnhancers,
  parameters
};
