// @storybook/web-components@10.0.8/dist/entry-preview-docs.js
import {
  __name
} from "./_browser-chunks/chunk-JFJ5UJ7Q.js";

// src/entry-preview-docs.ts
import { SourceType as SourceType2 } from "storybook/internal/docs-tools";

// src/docs/sourceDecorator.ts
import { SourceType } from "storybook/internal/docs-tools";
import { render } from "lit";
import { emitTransformCode, useEffect } from "storybook/preview-api";
var LIT_EXPRESSION_COMMENTS = /<!--\?lit\$[0-9]+\$-->|<!--\??-->/g;
function skipSourceRender(context) {
  const sourceParams = context?.parameters.docs?.source;
  const isArgsStory = context?.parameters.__isArgsStory;
  if (sourceParams?.type === SourceType.DYNAMIC) {
    return false;
  }
  return !isArgsStory || sourceParams?.code || sourceParams?.type === SourceType.CODE;
}
__name(skipSourceRender, "skipSourceRender");
function sourceDecorator(storyFn, context) {
  const story = storyFn();
  const renderedForSource = context?.parameters.docs?.source?.excludeDecorators ? context.originalStoryFn(context.args, context) : story;
  let source;
  useEffect(() => {
    if (source) {
      emitTransformCode(source, context);
    }
  });
  if (!skipSourceRender(context)) {
    const container = window.document.createElement("div");
    if (renderedForSource instanceof DocumentFragment) {
      render(renderedForSource.cloneNode(true), container);
    } else {
      render(renderedForSource, container);
    }
    source = container.innerHTML.replace(LIT_EXPRESSION_COMMENTS, "");
  }
  return story;
}
__name(sourceDecorator, "sourceDecorator");

// src/entry-preview-docs.ts
var decorators = [sourceDecorator];
var parameters = {
  docs: {
    source: {
      type: SourceType2.DYNAMIC,
      language: "html"
    },
    story: { inline: true }
  }
};
export {
  decorators,
  parameters
};
