// @storybook/web-components@10.0.8/dist/_browser-chunks/chunk-CTBWQQZK.js
import {
  entry_preview_exports
} from "./chunk-NIAEHH3S.js";
import {
  __name
} from "./chunk-JFJ5UJ7Q.js";

// src/index.ts
import { global as global3 } from "@storybook/global";

// src/globals.ts
import { global } from "@storybook/global";
var { window: globalWindow } = global;
globalWindow.STORYBOOK_ENV = "web-components";

// src/framework-api.ts
import { global as global2 } from "@storybook/global";
function isValidComponent(tagName) {
  if (!tagName) {
    return false;
  }
  if (typeof tagName === "string") {
    return true;
  }
  throw new Error('Provided component needs to be a string. e.g. component: "my-element"');
}
__name(isValidComponent, "isValidComponent");
function isValidMetaData(customElements) {
  if (!customElements) {
    return false;
  }
  if (customElements.tags && Array.isArray(customElements.tags) || customElements.modules && Array.isArray(customElements.modules)) {
    return true;
  }
  throw new Error(`You need to setup valid meta data in your config.js via setCustomElements().
    See the readme of addon-docs for web components for more details.`);
}
__name(isValidMetaData, "isValidMetaData");
function setCustomElements(customElements) {
  global2.__STORYBOOK_CUSTOM_ELEMENTS__ = customElements;
}
__name(setCustomElements, "setCustomElements");
function setCustomElementsManifest(customElements) {
  global2.__STORYBOOK_CUSTOM_ELEMENTS_MANIFEST__ = customElements;
}
__name(setCustomElementsManifest, "setCustomElementsManifest");
function getCustomElements() {
  return global2.__STORYBOOK_CUSTOM_ELEMENTS__ || global2.__STORYBOOK_CUSTOM_ELEMENTS_MANIFEST__;
}
__name(getCustomElements, "getCustomElements");

// src/portable-stories.ts
import {
  setProjectAnnotations as originalSetProjectAnnotations,
  setDefaultProjectAnnotations
} from "storybook/preview-api";
function setProjectAnnotations(projectAnnotations) {
  setDefaultProjectAnnotations(entry_preview_exports);
  return originalSetProjectAnnotations(
    projectAnnotations
  );
}
__name(setProjectAnnotations, "setProjectAnnotations");

// src/index.ts
var { window, EventSource } = global3;
if (typeof module !== "undefined" && module?.hot?.decline) {
  module.hot.decline();
  const hmr = new EventSource("__webpack_hmr");
  hmr.addEventListener("message", /* @__PURE__ */ __name(function fullPageReload(event) {
    try {
      const { action } = JSON.parse(event.data);
      if (action === "built") {
        window.location.reload();
      }
    } catch (error) {
    }
  }, "fullPageReload"));
}

export {
  isValidComponent,
  isValidMetaData,
  setCustomElements,
  setCustomElementsManifest,
  getCustomElements,
  setProjectAnnotations
};
