// @storybook/web-components@10.0.8/dist/_browser-chunks/chunk-NIAEHH3S.js
import {
  __export,
  __name
} from "./chunk-JFJ5UJ7Q.js";

// src/entry-preview.ts
var entry_preview_exports = {};
__export(entry_preview_exports, {
  parameters: () => parameters,
  render: () => render,
  renderToCanvas: () => renderToCanvas
});

// src/render.ts
import { global } from "@storybook/global";
import { render as litRender } from "lit";
import { isTemplateResult } from "lit/directive-helpers.js";
import { simulateDOMContentLoaded, simulatePageLoad } from "storybook/preview-api";
import { dedent } from "ts-dedent";
var { Node } = global;
var render = /* @__PURE__ */ __name((args, context) => {
  const { id, component } = context;
  if (!component) {
    throw new Error(
      `Unable to render story ${id} as the component annotation is missing from the default export`
    );
  }
  const element = document.createElement(component);
  Object.entries(args).forEach(([key, val]) => {
    element[key] = val;
  });
  return element;
}, "render");
function renderToCanvas({ storyFn, kind, name, showMain, showError, forceRemount }, canvasElement) {
  const element = storyFn();
  showMain();
  if (isTemplateResult(element)) {
    if (forceRemount || !canvasElement.querySelector('[id="root-inner"]')) {
      canvasElement.innerHTML = '<div id="root-inner"></div>';
    }
    const renderTo = canvasElement.querySelector('[id="root-inner"]');
    litRender(element, renderTo);
    simulatePageLoad(canvasElement);
  } else if (typeof element === "string") {
    canvasElement.innerHTML = element;
    simulatePageLoad(canvasElement);
  } else if (element instanceof Node) {
    if (canvasElement.firstChild === element && !forceRemount) {
      return;
    }
    canvasElement.innerHTML = "";
    canvasElement.appendChild(element);
    simulateDOMContentLoaded();
  } else {
    showError({
      title: `Expecting an HTML snippet or DOM node from the story: "${name}" of "${kind}".`,
      description: dedent`
        Did you forget to return the HTML snippet from the story?
        Use "() => <your snippet or node>" or when defining the story.
      `
    });
  }
}
__name(renderToCanvas, "renderToCanvas");

// src/entry-preview.ts
var parameters = {
  renderer: "web-components"
};

export {
  render,
  renderToCanvas,
  parameters,
  entry_preview_exports
};
